pkgs/dartpad_ui/lib/main.dart

```diff
@@ -227,16 +229,18 @@ class DartPadMainPage extends StatefulWidget {
        );
 
   @override
-  State<DartPadMainPage> createState() => _DartPadMainPageState();
+  State<DartPadMainPage> createState() => DartPadMainPageState();
 }
 
-class _DartPadMainPageState extends State<DartPadMainPage>
+@visibleForTesting
+class DartPadMainPageState extends State<DartPadMainPage>
     with SingleTickerProviderStateMixin {
   late final AppModel appModel;
   late final AppServices appServices;
   late final SplitViewController mainSplitter;
   late final SplitViewController consoleSplitter;
   late final TabController tabController;
+  final initialized = Completer<void>();
```

Comment:
nit: this could be a private Completer that exposes a Future for it's public interface (other objects don't need the ability to complete this from the outside)

Reply:
I would keep it simple for readability. I do not think this protection is necessary.